File: neetcode.io/01-arrays/03-TwoSum/main.js
// Copyright (c) 2023 Furukawa, Atsushi <atsushifx@aglabo.com>
// 
// This software is released under the MIT License.
// https://opensource.org/licenses/MIT

/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
	dic = []
	for (let i = 0; i < nums.length - 1; i++) {
		const n = nums[i]
		let i1 = dic[target - n]
		if (i1 !== undefined) {
			return [i1, i];
		}
		dic[n] = i
	}
	console.log(dic)
	return [];
};

console.log(twoSum([2, 7, 11, 8], 9))

File: neetcode.io/01-arrays/03-TwoSum/ruby/main.rb
#
# Copyright (c) 2023 Furukawa, Atsushi <atsushifx@aglabo.com>
#
# This software is released under the MIT License.
# https://opensource.org/licenses/MIT
#

# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer[]}
def two_sum(nums, target)
    dic = {}
    nums.each_with_index do |n, i|
        if dic[target - n]
            return [dic[target - n], i]
        end
        dic[n] = i
    end
    []
end

p two_sum([2, 7, 11, 15], 9)
p two_sum([3, 2, 4], 6)
p two_sum([3, 3], 6)
